src/options_app.c

```diff
@@ -649,11 +537,86 @@ static BOOL ov54_021E5DBC(OptionsApp_Data *data) {
 
 static void OptionsApp_SetupBgConfig(OptionsApp_Data *data) {
     data->bgConfig = BgConfig_Alloc(data->heapId);
-    GraphicsModes modes = sOptionsAppGraphicsModes;
+    GraphicsModes modes = {
+        .dispMode = GX_DISPMODE_GRAPHICS,
+        .bgMode = GX_BGMODE_0,
+        .subMode = GX_BGMODE_0,
+        ._2d3dMode = GX_BG0_AS_2D,
+    };
     SetBothScreensModesAndDisable(&modes);
 
-    BgTemplate templates[5];
-    templates = sOptionsAppBgTemplates;
+    BgTemplate templates[5] = {
+        {
```

Comment:
same... these inline structs make these funcs very messy

Reply:
this fixes a nonstandard assignment to array

Reply:
this is understandable, however the structs should be outside the func imo, so long as this matches